### ISSUE TITLE
Compiler panics when compiling a project in OneDrive

### ISSUE THREAD

**Issue by HeroicosHM** [rust-lang/rust#49895]
## Issue
So I have been getting into rust recently, came across this error, and nobody I ask has any idea how to work with it. I am working on chapter 3.3, dealing with function, which I understand the logic of because it is vey similar to other languages.  All my code is run using:
```rust
cargo run
```

The first set of code I run is:
```rust
fn main() {
    another_function(5);
}

fn another_function(x: i32) {
    println!("The value of x is: {}", x);
}
```
Then I try to run:
```rust
fn main() {
    another_function(5, 6);
}

fn another_function(x: i32, y: i32) {
    println!("The value of x is: {}", x);
    println!("The value of y is: {}", y);
}
```

The first one works fine, but then when I try to run the second set of code (again using "cargo run"), it returns this error:
![rusterror](https://user-images.githubusercontent.com/32232419/38658305-66fc457e-3dd9-11e8-8695-5a9b10a8e418.PNG)

## Updates
About an hour later, I have tried many different methods to no avail, including completely uninstalling rust, using "rustup self uninstall" and then reinstalling.

I can get it to work again by deleting the target folder, then running "cargo clean" ("cargo clean" does not run until the target folder is deleted), and then using "cargo run", but then as soon as any new changes are made it breaks again.

(Also I apologize for any lack of proper formatting in the error message, I am new to github as well.)
## Meta

rustc 1.25.0 (84203cac6 2018-03-25)
binary: rustc
commit-hash: 84203cac67e65ca8640b8392348411098c856985
commit-date: 2018-03-25
host: x86_64-pc-windows-msvc
release: 1.25.0
LLVM version: 6.0

**Comment by pietroalbini** [comment]
cc @retep998

**Comment by cswl** [comment]
We discussed this in Discord server.. then I noticed in the screenshot path the OneDrive
OneDrive might be locking the files during sync...

While this may affect only a small numbers of users... 
The error message could be improved.. instead of panicking..

**Comment by pietroalbini** [comment]
Yeah, the compiler should never panic, this is still a (diagnostics) bug.

**Comment by HeroicosHM** [comment]
Oh alright, my mistake.

**Comment by pitdicker** [comment]
Started investigating, and I think the issue was with how OneDrive started using a new method involving reparse points for dictionaries, that confused the standard library.

I am pretty sure this was fixed by https://github.com/rust-lang/rust/pull/47956, but haven't tried to make a setup to test it. Issues with OneDrive started in december 2017 (https://github.com/rust-lang/rust/issues/46484), and the fix landed in Rust 1.26 (this issue was with Rust 1.25).

Proposing to close.

**Comment by steveklabnik** [comment]
Triage: with no comments since the previous post, I'm going to close this issue. If anyone can reproduce this bug today, please let me know and we'll re-open. Thank you!
